src/utils/promise.ts

```diff
@@ -0,0 +1,20 @@
+type AsyncFunc<T, R> = (item: T) => Promise<R>;
```

Comment:
declaring this type gives no gain in readability and DRY, you should inline the type below